internal/job/vault_seal_watcher.go

```diff
@@ -91,43 +92,70 @@ func (v *VaultSealWatcher) handleUnsealForHAVault() error {
 
 		vaultClients = append(vaultClients, vc)
 	}
-
-	allSealed, err := v.isAllNodesSealed(vaultClients)
-	if err != nil {
-		return err
-	}
-
-	if allSealed {
-		v.log.Info("vault is sealed for all nodes")
-		err = vaultClients[0].Unseal()
-		if err != nil {
-			return fmt.Errorf("failed to unseal vault for leader node, %v", err)
-		}
-
-		err = vaultClients[1].JoinRaftCluster(v.conf.NodeAddresses[0])
-		if err != nil {
-			return fmt.Errorf("failed to join the HA cluster by 2nd node, %v", err)
-		}
-
-		err = vaultClients[1].Unseal()
-		if err != nil {
-			return fmt.Errorf("failed to unseal vault for 2nd node, %v", err)
-		}
-
-		err = vaultClients[2].JoinRaftCluster(v.conf.NodeAddresses[0])
+	for _, vc := range vaultClients {
+		sealed, err := vc.IsVaultSealed()
 		if err != nil {
-			return fmt.Errorf("failed to join the HA cluster by 3rd node, %v", err)
+			return err
 		}
+		if sealed {
+			switch vc {
+			case vaultClients[0]:
+				err = vc.Unseal()
+				if err != nil {
+					return fmt.Errorf("failed to unseal vault for leader node, %v", err)
+				}
+			default:
+				err = vc.JoinRaftCluster(v.conf.NodeAddresses[0])
+				if err != nil {
+					return fmt.Errorf("failed to join the HA cluster, %v", err)
+				}
+
+				err = vc.Unseal()
+				if err != nil {
+					return fmt.Errorf("failed to unseal vault, %v", err)
+				}
+			}
```

Comment:
this same code as before, only thing is partial sealed pods to be handled.

revert to previous, in this else case, pod unseal for 2nd and 3rd to be handled.

} else {
		v.log.Info("some vault nodes are sealed")
	}